### PR TITLE
Update test_average_precision.py

### PR DESCRIPTION
**Pull request by puhuk**
Description: Update `_test_distrib_integration_list_of_tensors_or_numbers` to test with different rank

Check list:

- [x] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

**Comment by puhuk** [comment]
A question :
When I test with 
`CUDA_VISIBLE_DEVICES="" pytest --dist=each --tx $WORLD_SIZE*popen//python=python -vvv tests/ignite/contrib/metrics/test_average_precision.py`, it works

But when I try with below, it runs no tests.
`CUDA_VISIBLE_DEVICES="" pytest --dist=each --tx $WORLD_SIZE*popen//python=python -vvv tests/ignite/contrib/metrics/test_average_precision.py -m distribute`

File: tests/ignite/contrib/metrics/test_average_precision.py [hunk 1]
```diff
@@ -215,13 +215,12 @@ def get_test_cases():
...
     rank = idist.get_rank()
-    torch.manual_seed(12)
     n_iters = 80
-    s = 16
+    batch_size = 16
     n_classes = 2
-    offset = n_iters * s
 
     def _test(y_preds, y_true, n_epochs, metric_device, update_fn):
+        torch.manual_seed(12 + rank)
```
Comment (vfdev-5) [inline tests/ignite/contrib/metrics/test_average_precision.py:223]: This is incorrect as `y_preds, y_true` are already randomly generated

File: tests/ignite/contrib/metrics/test_average_precision.py [hunk 2]
```diff
@@ -240,24 +241,25 @@ def _test(y_preds, y_true, n_epochs, metric_device, update_fn):
...
-            y_preds = torch.randint(0, n_classes, size=(offset * idist.get_world_size(), 10)).to(device)
+            y_true = torch.randint(0, n_classes, size=(n_iters * batch_size, 10)).to(device)
+            y_preds = torch.randint(0, n_classes, size=(n_iters * batch_size, 10)).to(device)
 
             def update_fn(engine, i):
                 return (
-                    y_preds[i * s + rank * offset : (i + 1) * s + rank * offset, :],
-                    y_true[i * s + rank * offset : (i + 1) * s + rank * offset, :],
+                    y_preds[i * batch_size * n_iters : (i + 1) * batch_size * n_iters, :],
+                    y_true[i * batch_size * n_iters : (i + 1) * batch_size * n_iters, :],
```
Comment (vfdev-5) [inline tests/ignite/contrib/metrics/test_average_precision.py:262]: Why `i * batch_size * n_iters` here and `i * batch_size` on Line 251 ?

Comment (puhuk) [reply tests/ignite/contrib/metrics/test_average_precision.py]: I mistakenly leave `n_iters`. I send again :)

**Review by vfdev-5 — APPROVED** [review]
Thanks

**Pull request merged**